### DISCUSSION TITLE
Extending components

### DISCUSSION THREAD

**Discussion by rsktash** [badoo/Decompose#49] [Ideas]
Is it a right decision to inject feature module logic dependencies into component store?

I have a default module and some extra feature modules. These extra features make some changes to the ui. I don’t want to duplicate shared logic and ui

![image](https://user-images.githubusercontent.com/36570577/105622207-4214cd00-5e31-11eb-9065-b6ad22400e90.png)
![image](https://user-images.githubusercontent.com/36570577/105622209-47721780-5e31-11eb-9305-94008e3846d8.png)

**Comment by arkivanov** [comment]
@rustamsmax  Is it somehow related to Decompose library? Just in case, MVIKotlin also has Descussions [enabled](https://github.com/arkivanov/MVIKotlin/discussions). There is also dedicated Slack channel [#mvikotlin](https://kotlinlang.slack.com/archives/C01403U1ZGW).

**Reply by rsktash** [reply]
Oh. Sorry. It's related to mvikotlin.

**Reply by arkivanov** [reply]
Thanks! I see multiple issues in the provided code, would nice to discuss them, as well as your main question.
